### DISCUSSION TITLE
Separate access to the data, depending on client's IP, Lizardfs-style

### DISCUSSION THREAD

**Discussion by kulakkulak25** [seaweedfs/seaweedfs#2415] [Q&A]
Searching the way to separate access to the data stored in seaweedfs, depending on IP adresses of clients.
In Lizardfs you can set this kind of rules the following way:
```
${IP_01}   /statistics            rw,alldirs,maproot=0,mintrashtime=0,maxtrashtime=300
${IP_02}   /backup                rw,alldirs,maproot=0
${IP_03}   /external              ro,alldirs,maproot=0
${IP_04}   /internal              rw,alldirs,maproot=0
```

For seaweedfs I have not found the method to control access depending on client's IP.

For seaweedfs I've found the following:
1. `weed volume -whiteList`;
2. firewall on volume servers;
3. `mount -volumeServerAccess=direct`;
4. `mount -filer.path=/separate/directory/for/this/user`;
5. use -disk=${tag} (for both weed volume and weed mount) not only for implementation of multi-tier, but for dividing data in order to provide separate access. 

But it seems not enough.
For example
	
A hacker/intruder who got root-access to client's node, would be able to remount seaweedfs storage (with option -filer.path=/) and get access to all the data.
The access limitations, set via volume -whiteList or iptables/nftables on volume-servers, can be violated explicitly using option -volumeServerAccess=filerProxy.

It means, `mount weed -filer.path=/project01/backup/pgdumps/ -volumeServerAccess=direct`
can be replaced with `mount weed -filer.path=/ -volumeServerAccess=filerProxy`
and thus access limitations can be violated.

What are the ways to divide data access rights depending on IP address for seaweedfs, similar to those ways of Lizardfs?

**Comment by chrislusf** [comment]
Currently there are no good mechanism to achieve this.

One way is to enable TLS following gRPC https://github.com/chrislusf/seaweedfs/wiki/Security-Overview , so that only servers with valid TLS certs can work with the SeaweedFS cluster. This of course can not stop hackers who can get root access of the node.
